setup.sh

```diff
@@ -10,8 +10,20 @@ if [ $EUID -ne 0 ]; then
   exit 1
 fi
 
-if git submodule status --recursive | grep -q "^-"; then
-  git submodule update --init --recursive
+tmpfile=$(mktemp)
+# any error messages from this command will stand out
+# more clearly when run as a separate command rather than
+# being piped
+git submodule status --recursive > "$tmpfile"
+
+if grep -q "^-" "$tmpfile"; then
+  sudo -u $SUDO_USER git submodule update --init --recursive
+elif grep -q "^+" "$tmpfile"; then
+  # Make it easy for users who are not hacking ORFS to do the right thing,
+  # run with current submodules, at the cost of having ORFS
+  # hackers disable this test manually when hacking setup.sh
+  echo "Submodules are not up to date, run 'git submodule update --recursive' and try again"
+  exit 1
```

Comment:
I'm not enthusiastic about this being a hard error but we'll see how it goes.

Reply:
I would rather err on the side of doing nothing with an actionable message, than damage...

Reply:
Though: this PR is evidence of silent suffering... I think we'd have to peer over colleagues shoulders to pick up the problems they are seeing when occasionally hacking ORFS....